x-pack/plugins/triggers_actions_ui/public/application/sections/rule_details/components/rule_details.tsx

```diff
@@ -340,175 +337,63 @@ export const RuleDetails: React.FunctionComponent<RuleDetailsProps> = ({
-                      } else {
-                        setIsEnabled(true);
-                        await enableRule(rule);
-                      }
-                      requestRefresh();
-                      setIsEnabledUpdating(false);
-                    }}
-                    label={
-                      <FormattedMessage
-                        id="xpack.triggersActionsUI.sections.ruleDetails.collapsedItemActons.enableTitle"
-                        defaultMessage="Enable"
-                      />
-                    }
-                  />
-                )}
-              </EuiFlexItem>
-              <EuiFlexItem grow={false}>
-                {isMutedUpdating ? (
-                  <EuiFlexGroup>
-                    <EuiFlexItem>
-                      <EuiLoadingSpinner size="m" />
-                    </EuiFlexItem>
-
-                    <EuiFlexItem>
-                      <EuiText size="s">
-                        <FormattedMessage
-                          id="xpack.triggersActionsUI.sections.ruleDetails.collapsedItemActons.muteLoadingTitle"
-                          defaultMessage="Mute"
-                        />
-                      </EuiText>
-                    </EuiFlexItem>
-                  </EuiFlexGroup>
-                ) : (
-                  <EuiSwitch
-                    name="mute"
-                    checked={isMuted}
-                    disabled={!canSaveRule || !isEnabled || !ruleType.enabledInLicense}
-                    data-test-subj="muteSwitch"
-                    onChange={async () => {
-                      setIsMutedUpdating(true);
-                      if (isMuted) {
-                        setIsMuted(false);
-                        await unmuteRule(rule);
-                      } else {
-                        setIsMuted(true);
-                        await muteRule(rule);
-                      }
-                      requestRefresh();
-                      setIsMutedUpdating(false);
-                    }}
-                    label={
-                      <FormattedMessage
-                        id="xpack.triggersActionsUI.sections.ruleDetails.collapsedItemActons.muteTitle"
-                        defaultMessage="Mute"
-                      />
-                    }
-                  />
-                )}
-              </EuiFlexItem>
-            </EuiFlexGroup>
+            <RuleStatusDropdown
+              disableRule={async () => await disableRule(rule)}
+              enableRule={async () => await enableRule(rule)}
+              snoozeRule={async (snoozeEndTime: string | -1) =>
+                await snoozeRule(rule, snoozeEndTime)
+              }
+              unsnoozeRule={async () => await unsnoozeRule(rule)}
+              item={rule as RuleTableItem}
+              onRuleChanged={requestRefresh}
+              direction="row"
+              isEditable={hasEditButton}
+              previousSnoozeInterval={null}
```

Comment:
do we need to specifically pass in null here?

Reply:
Yes because we are not going to backport this feature for 8.2, so I will have to remove it. But I think that we should also move the code out of the rule list page to `RuleStatusDropdown`, therefore we are not duplicating this functionality everywhere we used this component. We will follow up with a new PR.